longtailstorelib/remotestore.go

```diff
@@ -990,14 +988,6 @@ func contentIndexWorker(
 		return nil
 	}
 
-	for len(blockIndexMessages) > 0 {
```

Comment:
This logic is what broke - we now wait in the proper way for all blockIndexMessages to be picked up using the "more" flag.
The other end of the code does not wait for a response for the messages to be picked up and the old len(blockIndexMessages) check is not a safe way to get all the messages.